src/librustdoc/clean/types.rs

```diff
@@ -264,12 +264,9 @@ impl Item {
     }
 
     crate fn is_crate(&self) -> bool {
-        matches!(
-            *self.kind,
-            StrippedItem(box ModuleItem(Module { is_crate: true, .. }))
-                | ModuleItem(Module { is_crate: true, .. })
-        )
+        self.is_mod() && self.def_id.index == CRATE_DEF_INDEX
```

Comment:
Am I correct in thinking that this should also work for crates other than the current crate, because all crates get `CRATE_DEF_INDEX` as their DefIndex?

Reply:
Correct, the DefIndex doesn't care about which crate it comes from. If you look at DefId it's actually just a (CrateNum, DefIndex) pair: https://doc.rust-lang.org/nightly/nightly-rustc/rustc_hir/def_id/struct.DefId.html

Reply:
The reason I added the `self.is_mod()` check is because I think some of the fake DefIds are actually the crate DefId instead of a 'proper' unique ID. I could remove it someone double checks that all the fake DefIds are unique.

Reply:
> The reason I added the `self.is_mod()` check is because I think some of the fake DefIds are actually the crate DefId instead of a 'proper' unique ID. I could remove it someone double checks that all the fake DefIds are unique.

That would be weird, but I wouldn't put it past rustdoc ;)

It seems fine to keep the `self.is_mod()` check.